### PR TITLE
Rewrite Dependency Injection Chapter

### PR DESCRIPTION
**Pull request by tomwarwick**

### CODE REVIEW

File: Documentation/ApiOverview/DependencyInjection/Index.rst [hunk 1]
```diff
@@ -9,33 +8,400 @@ Dependency injection
...
+   :local:
+
+Introduction
+============
+
+The title of this chapter is "dependency injection" (DI), but the scope is a bit
+broader: In general, this chapter is about TYPO3 object lifecycle management,
+how to obtain objects, with one sub-part of it being dependency injection.
+
+The underlying interfaces are based on the PHP-FIG standard
```
Comment (susannemoog) [inline Documentation/ApiOverview/DependencyInjection/Index.rst:24]: The general structure should be more in line with the needs of readers:
- first: see how to use it quickly
- then: add explanations / implementation detail
- then: add read more section for further info
- last: add history

**Comment by lolli42** [comment]
not interested anymore. don't push again.

**Comment by jonaseberle** [comment]
I still think this is very interesting!

**Comment by Lefaux** [comment]
grow the hell up

**Comment by jonaseberle** [comment]
@Lefaux what does that mean? Please elaborate. Have I done something wrong?

**Comment by Lefaux** [comment]
My comment was for Lolli.
The content in the change is useful for the documentation, it needs lingual polishing and thinning out of opinions.
That's why the workflow using PRs is in place.

Reacting to having to align with that workflow by closing down the entire change is what I consider childish - hence the request to grow up.

**Pull request closed without merging**